core/server/src/main/java/alluxio/worker/block/BlockMasterSync.java

```diff
@@ -73,6 +77,10 @@
   /** Last System.currentTimeMillis() timestamp when a heartbeat successfully completed. */
   private long mLastSuccessfulHeartbeatMs;
 
+  /** Map from a block Id to whether it has been removed successfully. */
+  @GuardedBy("itself")
+  private final Map<Long, Boolean> mRemovingBlockIdToFinished;
```

Comment:
It seems like the current implementation is using this map like a set. Is there a reason why this is a map and not a set? Couldn't this "set" simply track which blockId removals are "in progress"? I'm not sure what the extra bit of information (the boolean value) provides?